pkg/controller/load_configuration_test.go

```diff
@@ -0,0 +1,238 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestLoadFromConfigMap(t *testing.T) {
+	restore := test.SetEnvVarAndRestore(t, "WATCH_NAMESPACE", "toolchain-member-operator")
+	defer restore()
+
+	t.Run("configMap not found", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config")
+		defer restore()
+
+		cl := test.NewFakeClient(t)
+
+		// when
+		err := LoadFromConfigMap("MEMBER_OPERATOR", "MEMBER_OPERATOR_CONFIG_MAP_NAME", cl)
+
+		// then
+		require.NoError(t, err)
+	})
+	t.Run("no config name set", func(t *testing.T) {
+		// given
+		data := map[string]string{
+			"super-special-key": "super-special-value",
+		}
+		cl := test.NewFakeClient(t, createConfigMap("test-config", "toolchain-host-operator", data))
+
+		// when
+		err := LoadFromSecret("HOST_OPERATOR", "HOST_OPERATOR_CONFIG_MAP_NAME", cl)
+
+		// then
+		require.NoError(t, err)
+
+		// test that the secret was not found since no secret name was set
+		testTest := os.Getenv("HOST_OPERATOR_SUPER_SPECIAL_KEY")
+		assert.Equal(t, "", testTest)
+	})
+	t.Run("cannot get configmap", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config")
+		defer restore()
+
+		data := map[string]string{
+			"test-key-one": "test-value-one",
+		}
+		cl := test.NewFakeClient(t, createConfigMap("test-config", "toolchain-host-operator", data))
+
+		cl.MockGet = func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+			return errors.New("oopsie woopsie")
+		}
+
+		// when
+		err := LoadFromConfigMap("MEMBER_OPERATOR", "MEMBER_OPERATOR_CONFIG_MAP_NAME", cl)
+
+		// then
+		require.Error(t, err)
+		assert.Equal(t, "oopsie woopsie", err.Error())
+
+		// test env vars are parsed and created correctly
+		testTest := os.Getenv("MEMBER_OPERATOR_TEST_KEY_ONE")
+		assert.Equal(t, testTest, "")
+	})
+	t.Run("env overwrite", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config")
```

Comment:
it's just a detail, but we should also set and restore the `MEMBER_OPERATOR_TEST_KEY` var
```suggestion
		restore := test.SetEnvVarsAndRestore(t,
			test.Env("MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config"),
			test.Env("MEMBER_OPERATOR_TEST_KEY", ""))
```

Reply:
fixed here: 8cba0fc9d72606ec22c4ef1d91cab9288ab19460